tests/core/test_model.py

```diff
@@ -157,3 +162,23 @@ def test_context_size_caching_works() -> None:
     _cached_context_size.cache_clear()
     different_result = another_model_instance.context_size
     assert context_size is not different_result
+
+
+def test_chat_model_can_produce_chat_prompt() -> None:
+    client = DummyModelClient()  # type: ignore
+    model = Llama3ChatModel("llama-3.1-8b-instruct", client)
+    messages = [
+        Message(role="system", content="You are a nice assistant."),
+        Message(role="user", content="What's 2+2?"),
+    ]
+    response_prefix = "The answer is"
+
+    prompt = model.to_chat_prompt(messages=messages, response_prefix=response_prefix)
+
+    assert isinstance(prompt.items[0], Text)
+
+    text_in_prompt = prompt.items[0].text
+
+    assert messages[0].content in text_in_prompt
```

Comment:
Could we explicitly check the final result of the prompt? In the end, the function's desired behavior is to generate the correct template format. Usually `in` can lead to mistakes :)

Reply:
done